src/bundle.ts

```diff
@@ -95,3 +108,52 @@ export class BuildBundler extends Transform {
         });
   }
 }
+
+export type ResolveFileCallback = (a: string) => void;
+export type RejectFileCallback = (err: Error) => void;
+export type DeferredFileCallbacks = {
+  resolve: ResolveFileCallback; reject: RejectFileCallback;
+};
+
+class StreamLoader implements UrlLoader {
+  bundler: BuildBundler;
+
+  // Store files that have not yet entered the Analyzer stream here.
+  // Later, when the file is seen, the DeferredFileCallback can be
+  // called with the file contents to resolve its loading.
+  deferredFiles = new Map<string, DeferredFileCallbacks>();
```

Comment:
Add a comment that mentions whose responsibility it is to go through these and resolve or reject them.